crates/transpiler/src/passes/convert_discrete_rotations.rs

```diff
@@ -0,0 +1,369 @@
+        }
+        (StandardGate::RY, Some(6)) => {
+            discrete_sequence.push(StandardGate::H);
+            discrete_sequence.push(StandardGate::Z);
+            -1.0 * PI
+        }
+        (StandardGate::RY, Some(7)) => {
+            discrete_sequence.push(StandardGate::SX);
+            discrete_sequence.push(StandardGate::Tdg);
+            discrete_sequence.push(StandardGate::SXdg);
+            -7.0 * PI8
+        }
+        (StandardGate::RY, Some(8)) => {
+            discrete_sequence.push(StandardGate::I);
+            -1.0 * PI
+        }
+        (StandardGate::RY, Some(9)) => {
+            discrete_sequence.push(StandardGate::SX);
+            discrete_sequence.push(StandardGate::T);
+            discrete_sequence.push(StandardGate::SXdg);
+            -9.0 * PI8
+        }
+        (StandardGate::RY, Some(10)) => {
+            discrete_sequence.push(StandardGate::Z);
+            discrete_sequence.push(StandardGate::H);
+            -1.0 * PI
+        }
+        (StandardGate::RY, Some(11)) => {
+            discrete_sequence.push(StandardGate::SX);
+            discrete_sequence.push(StandardGate::T);
+            discrete_sequence.push(StandardGate::S);
+            discrete_sequence.push(StandardGate::SXdg);
+            -11.0 * PI8
+        }
+        (StandardGate::RY, Some(12)) => {
+            discrete_sequence.push(StandardGate::Y);
+            -12.0 * PI8
+        }
+        (StandardGate::RY, Some(13)) => {
+            discrete_sequence.push(StandardGate::Y);
+            discrete_sequence.push(StandardGate::SX);
+            discrete_sequence.push(StandardGate::T);
+            discrete_sequence.push(StandardGate::SXdg);
+            -13.0 * PI8
+        }
+        (StandardGate::RY, Some(14)) => {
+            discrete_sequence.push(StandardGate::H);
+            discrete_sequence.push(StandardGate::Z);
+            -0.0
+        }
+        (StandardGate::RY, Some(15)) => {
+            discrete_sequence.push(StandardGate::SX);
+            discrete_sequence.push(StandardGate::Tdg);
+            discrete_sequence.push(StandardGate::SXdg);
+            -15.0 * PI8
+        }
+        _ => {
+            return None;
+        }
+    };
+
+    Some((discrete_sequence, global_phase))
+}
+
+#[pyfunction]
+#[pyo3(name = "discretize_rotations")]
+pub fn run_discretize_rotations(dag: &mut DAGCircuit, approximation_degree: f64) -> PyResult<()> {
+    let op_counts = dag.get_op_counts();
+    let tol = MINIMUM_TOL.max(1.0 - approximation_degree);
+
+    // Skip the pass if there are no RX/RY/RZ rotation gates.
+    if op_counts
+        .keys()
+        .all(|k| !ROTATION_GATE_NAMES.contains(&k.as_str()))
+    {
+        return Ok(());
+    }
+
+    // Iterate over nodes in the DAG and collect nodes that are of the form
+    // RX/RY/RZ with an angle that is a multiply of 2*pi/8
+    let mut candidates: Vec<(NodeIndex, StandardGate, f64)> = Vec::new();
+
+    for (node_index, inst) in dag.op_nodes(true) {
```

Comment:
The boolean argument to `op_nodes` specifies whether we want to include directives or not. Since the pass does not change directives, we can set this argument to `false`.

```suggestion
    for (node_index, inst) in dag.op_nodes(false) {
```

Reply:
done in [066ded3](https://github.com/Qiskit/qiskit/pull/15321/commits/066ded3f9e884d07037336a87d969f43cb54c8eb)